cpp/examples/tracing.dox

```diff
@@ -0,0 +1,74 @@
+// -*-markdown-*-
+// NOTE: doxygen can include markdown pages directly but there seems to be a bug
+// that shows messed-up line numbers in \skip \until code extracts. This file
+// is markdown wrapped in a doxygen comment - which works. The file is best viewed/edited
+// as markdown.
+
+/**
+
+@page tracing Tracing
+
+## Usage:
+
+Minimum requirements to enable tracing in proton:
+
+1. Include `<proton/tracing.hpp>`  
+2. Call `proton::initOpenTelemetryTracer()` with a service-name as an argument.
+
+
+A working example of distributed tracing using broker, server and client examples.
+
```

Comment:
Documentation for tracing is here having instructions for running the examples.
This will be shown on a separate page named Tracing.